xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -796,37 +785,29 @@ CGUIFontTTF::Character* CGUIFontTTF::GetCharacter(character_t chr, FT_UInt glyph
   int startIndex = low;
 
   // increase the size of the buffer if we need it
-  if (m_numChars >= m_maxChars)
-  { // need to increase the size of the buffer
-    Character* newTable = new Character[m_maxChars + CHAR_CHUNK];
-    if (m_char)
-    {
-      memcpy(newTable, m_char, low * sizeof(Character));
-      memcpy(newTable + low + 1, m_char + low, (m_numChars - low) * sizeof(Character));
-      delete[] m_char;
-    }
-    m_char = newTable;
-    m_maxChars += CHAR_CHUNK;
+  if (m_char.size() == m_char.capacity())
+  {
+    m_char.reserve(m_char.capacity() + CHAR_CHUNK);
     startIndex = 0;
   }
-  else
-  { // just move the data along as necessary
-    memmove(m_char + low + 1, m_char + low, (m_numChars - low) * sizeof(Character));
-  }
+
   // render the character to our texture
   // must End() as we can't render text to our texture during a Begin(), End() block
   unsigned int nestedBeginCount = m_nestedBeginCount;
   m_nestedBeginCount = 1;
   if (nestedBeginCount)
     End();
-  if (!CacheCharacter(glyphIndex, style, m_char + low))
+
+  m_char.emplace(m_char.begin() + low);
+  if (!CacheCharacter(glyphIndex, style, m_char.data() + low))
   { // unable to cache character - try clearing them all out and starting over
     CLog::LogF(LOGDEBUG, "Unable to cache character. Clearing character cache of {} characters",
-               m_numChars);
+               m_char.size());
     ClearCharacterCache();
     low = 0;
     startIndex = 0;
-    if (!CacheCharacter(glyphIndex, style, m_char))
+    m_char.resize(1);
```

Comment:
Why 1, not  CHAR_CHUNK?

Reply:
`capacity()` is already `CHAR_CHUNK` here because three lines before is called `ClearCharacterCache();`

Here is inserted first element in `m_char` and `size()` needs to be 1 to reflect current number of elements.

Perhaps is less confusing use:  `m_char.emplace(m_char.begin());` although functionally is equivalent. I changed to that.